app/test/task/fake_time.dart

```diff
@@ -252,6 +252,11 @@ class _FakeTime extends FakeTime {
 
     // Trigger the callback for the pending timer.
     for (final triggeredTimer in triggeredTimers) {
+      // Skip running periodic timers if they have been cancelled.
+      // TODO: remove if the exception doesn't show up in flaky tests
+      if (triggeredTimer._isPeriodic && !triggeredTimer.isActive) {
+        throw Exception('Periodic timer is no longer active when running it.');
```

Comment:
```suggestion
        throw StateError('Periodic timer is no longer active when running it.');
```
or `AssertionError` if this an internal error in the `fake_time` library.

Reply:
Oh, since then I was able to confirm that this is indeed the case for the flakes, updating the PR soon.